File: 412/main.ts
function fizzBuzz(n: number): string[] {
  const res: string[] = [];
  const RESULT_DICT: Record<number, string> = { 3: "Fizz", 5: "Buzz" };

  for (let i = 1; i <= n; i++) {
    let tempItem = "";
    for (const key in RESULT_DICT) {
      if (i % Number(key) === 0) {
        tempItem = tempItem + RESULT_DICT[key];
      }
    }

    if (tempItem === "") {
      tempItem = i.toString();
    }
    res.push(tempItem);
  }
  return res;
}

console.log(fizzBuzz(15))

File: 412/c++/main.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

template <typename S>
ostream &operator<<(ostream &os,
                    const vector<S> &vector)
{
    // Printing all the elements
    // using <<
    for (auto element : vector)
    {
        os << element << " ";
    }
    return os;
}

vector<string> fizzBuzz(int n)
{
    vector<string> result;

    map<int, string> RESULT_STRING{
        {3, "Fizz"},
        {5, "Buzz"},
    };

    for (int i = 1; i <= n; i++)
    {
        string tempItem = "";

        for (auto element : RESULT_STRING)
        {
            if (i % element.first == 0)
            {
                tempItem += element.second;
            }
        }

        if (tempItem == "")
        {
            tempItem = to_string(i);
        }

        result.push_back(tempItem);
    }

    return result;
}

int main()
{
    cout << fizzBuzz(15) << endl;
}